### PR TITLE
Anbangx/genomix scaffolding refactor

### PR DESCRIPTION
**Pull request by anbangx**
This is my refactor of scaffolding code (on github).

### CODE REVIEW

**Comment by anbangx** [comment]
@Anbang can you also take a look at this?

**Comment by JavierJia** [comment]
Well, I think it's better to finish the branches under retivels, because we already make a lot of comments there. Then  after the code review there, merge it with google code. 

Then @jakebiesinger , we can then pull the code from google code to github just for genomix/fullstack branch.

I think it's better to create newer branches here, and send the pull request on the new branches. This github is far more convenient than gg code.

**Comment by jakebiesinger** [comment]
@JavierJia yeah, I'm with you. I think @anbangx was just practicing here.

**Comment by anbangx** [comment]
Yep. I am trying to be familiar with it.

File: genomix/genomix-data/src/main/java/edu/uci/ics/genomix/config/GenomixJobConf.java [hunk 1]
```diff
@@ -384,16 +384,16 @@ private void fillMissingDefaults() {
         
         // scaffolding
         if (getInt(SCAFFOLDING_MIN_TRAVERSAL_LENGTH, -1) == -1)
-            setInt(SCAFFOLDING_MIN_TRAVERSAL_LENGTH, 20);
```
Comment (anbangx) [inline genomix/genomix-data/src/main/java/edu/uci/ics/genomix/config/GenomixJobConf.java:387]: Test comments in github

Comment (jakebiesinger) [inline genomix/genomix-data/src/main/java/edu/uci/ics/genomix/config/GenomixJobConf.java:387]: This is me responding to your comment from gmail.

## 

Jake Biesinger
Graduate Student
Xie Lab, UC Irvine

Comment (jakebiesinger) [inline genomix/genomix-data/src/main/java/edu/uci/ics/genomix/config/GenomixJobConf.java:387]: And this is me responding inline to your comment.

Comment (anbangx) [inline genomix/genomix-data/src/main/java/edu/uci/ics/genomix/config/GenomixJobConf.java:387]: OK. Which way do we use to respond other people's comments? By email or by inline?

Comment (jakebiesinger) [inline genomix/genomix-data/src/main/java/edu/uci/ics/genomix/config/GenomixJobConf.java:387]: As you can see, they go to the "correct" place either way.  Suit yourself.

## 

Jake Biesinger
Graduate Student
Xie Lab, UC Irvine

**Comment by jakebiesinger** [comment]
This is a comment that's NOT inline to the code.  These are meant to be more general about the whole process rather than individual lines.  Try responding to this via email and by commenting below-- you'll see the response show as a main comment, not as an inline one.

**Comment by jakebiesinger** [comment]
@Nan-Zhang you may want to come check this out and practice.

File: genomix/genomix-data/src/main/java/edu/uci/ics/genomix/config/GenomixJobConf.java (hunk 1 above)
Comment (JavierJia) [inline genomix/genomix-data/src/main/java/edu/uci/ics/genomix/config/GenomixJobConf.java:387]: far better than reitlet. I can't not even  spell it correct.

Comment (anbangx) [inline genomix/genomix-data/src/main/java/edu/uci/ics/genomix/config/GenomixJobConf.java:387]: Testing respond by email....

On Fri, Oct 18, 2013 at 12:00 PM, Jianfeng Jia notifications@github.comwrote:

> In
> genomix/genomix-data/src/main/java/edu/uci/ics/genomix/config/GenomixJobConf.java:
> 
> > @@ -384,16 +384,16 @@ private void fillMissingDefaults() {
> > 
> > ```
> >      // scaffolding
> >      if (getInt(SCAFFOLDING_MIN_TRAVERSAL_LENGTH, -1) == -1)
> > ```
> > -            setInt(SCAFFOLDING_MIN_TRAVERSAL_LENGTH, 20);
> 
> far better than reitlet. I can't not even spell it correct.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/uci-cbcl/genomix/pull/2/files#r7070896
> .

## 

Best Regards,

Anbang Xu

Comment (jakebiesinger) [inline genomix/genomix-data/src/main/java/edu/uci/ics/genomix/config/GenomixJobConf.java:387]: and again... this comment is from gmail.

## 

Jake Biesinger
Graduate Student
Xie Lab, UC Irvine

Comment (anbangx) [inline genomix/genomix-data/src/main/java/edu/uci/ics/genomix/config/GenomixJobConf.java:387]: inline comment from github

Comment (anbangx) [inline genomix/genomix-data/src/main/java/edu/uci/ics/genomix/config/GenomixJobConf.java:387]: Major comments from gmail.

On Fri, Oct 18, 2013 at 12:04 PM, Jake Biesinger
notifications@github.comwrote:

> In
> genomix/genomix-data/src/main/java/edu/uci/ics/genomix/config/GenomixJobConf.java:
> 
> > @@ -384,16 +384,16 @@ private void fillMissingDefaults() {
> > 
> > ```
> >      // scaffolding
> >      if (getInt(SCAFFOLDING_MIN_TRAVERSAL_LENGTH, -1) == -1)
> > ```
> > -            setInt(SCAFFOLDING_MIN_TRAVERSAL_LENGTH, 20);
> 
> and again... this comment is from gmail.
> … <#141ccf443853b147_>
> -- Jake Biesinger Graduate Student Xie Lab, UC Irvine
> On Fri, Oct 18, 2013 at 12:01 PM, anbangx notifications@github.com
> wrote: > In >
> genomix/genomix-data/src/main/java/edu/uci/ics/genomix/config/GenomixJobConf.java:
> 
> > > > @@ -384,16 +384,16 @@ private void fillMissingDefaults() { > > > > //
> > > > scaffolding > > if (getInt(SCAFFOLDING_MIN_TRAVERSAL_LENGTH, -1) == -1) > >
> > > > - setInt(SCAFFOLDING_MIN_TRAVERSAL_LENGTH, 20); > > Testing respond by
> > > >   email.... > … <#141ccf1d873cb3bb_> > On Fri, Oct 18, 2013 at 12:00 PM,
> > > >   Jianfeng Jia notifications@github.comwrote: In
> > > >   genomix/genomix-data/src/main/java/edu/uci/ics/genomix/config/GenomixJobConf.java:
> > > >   @@ -384,16 +384,16 @@ private void fillMissingDefaults() { > > //
> > > >   scaffolding > if (getInt(SCAFFOLDING_MIN_TRAVERSAL_LENGTH, -1) == -1) > -
> > > >   setInt(SCAFFOLDING_MIN_TRAVERSAL_LENGTH, 20); far better than reitlet. I
> > > >   can't not even spell it correct. — Reply to this email directly or view it
> > > >   on GitHubhttps://github.com/uci-cbcl/genomix/pull/2/files#r7070896 . --
> > > >   Best Regards, Anbang Xu — Reply to this email directly or view it on GitHub<
> > > >   https://github.com/uci-cbcl/genomix/pull/2/files#r7070937> .
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/uci-cbcl/genomix/pull/2/files#r7071007
> .

## 

Best Regards,

Anbang Xu

Comment (anbangx) [inline genomix/genomix-data/src/main/java/edu/uci/ics/genomix/config/GenomixJobConf.java:387]: Why my major comments go to inline comment if I directly reply the email?

On Fri, Oct 18, 2013 at 12:10 PM, Anbang Xu anbangx@gmail.com wrote:

> Major comments from gmail.
> 
> On Fri, Oct 18, 2013 at 12:04 PM, Jake Biesinger <notifications@github.com
> 
> > wrote:
> > 
> > In
> > genomix/genomix-data/src/main/java/edu/uci/ics/genomix/config/GenomixJobConf.java:
> > 
> > > @@ -384,16 +384,16 @@ private void fillMissingDefaults() {
> > > 
> > > ```
> > >      // scaffolding
> > >      if (getInt(SCAFFOLDING_MIN_TRAVERSAL_LENGTH, -1) == -1)
> > > ```
> > > -            setInt(SCAFFOLDING_MIN_TRAVERSAL_LENGTH, 20);
> > 
> > and again... this comment is from gmail.
> > … <#141ccfa22af8f2e5_141ccf443853b147_>
> > -- Jake Biesinger Graduate Student Xie Lab, UC Irvine
> > On Fri, Oct 18, 2013 at 12:01 PM, anbangx notifications@github.com
> > wrote: > In >
> > genomix/genomix-data/src/main/java/edu/uci/ics/genomix/config/GenomixJobConf.java:
> > 
> > > > > @@ -384,16 +384,16 @@ private void fillMissingDefaults() { > > > > //
> > > > > scaffolding > > if (getInt(SCAFFOLDING_MIN_TRAVERSAL_LENGTH, -1) == -1) > >
> > > > > - setInt(SCAFFOLDING_MIN_TRAVERSAL_LENGTH, 20); > > Testing respond by
> > > > >   email.... > … <#141ccf1d873cb3bb_> > On Fri, Oct 18, 2013 at 12:00 PM,
> > > > >   Jianfeng Jia notifications@github.comwrote: In
> > > > >   genomix/genomix-data/src/main/java/edu/uci/ics/genomix/config/GenomixJobConf.java:
> > > > >   @@ -384,16 +384,16 @@ private void fillMissingDefaults() { > > //
> > > > >   scaffolding > if (getInt(SCAFFOLDING_MIN_TRAVERSAL_LENGTH, -1) == -1) > -
> > > > >   setInt(SCAFFOLDING_MIN_TRAVERSAL_LENGTH, 20); far better than reitlet. I
> > > > >   can't not even spell it correct. — Reply to this email directly or view it
> > > > >   on GitHubhttps://github.com/uci-cbcl/genomix/pull/2/files#r7070896 .
> > > > >   -- Best Regards, Anbang Xu — Reply to this email directly or view it on
> > > > >   GitHubhttps://github.com/uci-cbcl/genomix/pull/2/files#r7070937 .
> > 
> > —
> > Reply to this email directly or view it on GitHubhttps://github.com/uci-cbcl/genomix/pull/2/files#r7071007
> > .
> 
> ## 
> 
> Best Regards,
> 
> Anbang Xu

## 

Best Regards,

Anbang Xu

Comment (jakebiesinger) [inline genomix/genomix-data/src/main/java/edu/uci/ics/genomix/config/GenomixJobConf.java:387]: You're replying to the wrong email.  You must respond to a major comment,
not an inline one, if you want it to be major.

I'm responding to you inline comment here.

## 

Jake Biesinger
Graduate Student
Xie Lab, UC Irvine

**Comment by jakebiesinger** [comment]
Now I'm responding to a major comment...

## 

Jake Biesinger
Graduate Student
Xie Lab, UC Irvine

**Comment by jakebiesinger** [comment]
this is a "major" comment, not inline.

## 

Jake Biesinger
Graduate Student
Xie Lab, UC Irvine

File: genomix/genomix-data/src/main/java/edu/uci/ics/genomix/type/DIR.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+    REVERSE((byte) (0b01 << 2)),
+    FORWARD((byte) (0b10 << 2));
+
+    public static final byte MASK = (byte) (0b11 << 2);
+    public static final byte CLEAR = (byte) (0b1110011);
+
+    private final byte val;
+
+    private DIR(byte val) {
```
Comment (jakebiesinger) [inline genomix/genomix-data/src/main/java/edu/uci/ics/genomix/type/DIR.java:15]: This is a new inline comment with markdown:

TODO list:
- [ ] get some more milk
- [x] make this code better

:smile_cat: 

this fixes #4 

I think the error was introduced in 4334f27f433484e6f93bdb055d9f70763ed4200a

**Comment by anbangx** [comment]
Try to response to major comments.

On Fri, Oct 18, 2013 at 10:55 AM, Jianfeng Jia notifications@github.comwrote:

> Well, I think it's better to finish the branches under retivels, because
> we already make a lot of comments there. Then after the code review there,
> merge it with google code.
> 
> Then @jakebiesinger https://github.com/jakebiesinger , we can then pull
> the code from google code to github just for genomix/fullstack branch.
> 
> I think it's better to create newer branches here, and send the pull
> request on the new branches. This github is far more convenient than gg
> code.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/uci-cbcl/genomix/pull/2#issuecomment-26616024
> .

## 

Best Regards,

Anbang Xu

File: genomix/genomix-data/src/main/java/edu/uci/ics/genomix/config/GenomixJobConf.java (hunk 1 above)
Comment (anbangx) [inline genomix/genomix-data/src/main/java/edu/uci/ics/genomix/config/GenomixJobConf.java:387]: Try to response to inline command by gmail

On Fri, Oct 18, 2013 at 11:43 AM, Jake Biesinger
notifications@github.comwrote:

> In
> genomix/genomix-data/src/main/java/edu/uci/ics/genomix/config/GenomixJobConf.java:
> 
> > @@ -384,16 +384,16 @@ private void fillMissingDefaults() {
> > 
> > ```
> >      // scaffolding
> >      if (getInt(SCAFFOLDING_MIN_TRAVERSAL_LENGTH, -1) == -1)
> > ```
> > -            setInt(SCAFFOLDING_MIN_TRAVERSAL_LENGTH, 20);
> 
> This is me responding to your comment from gmail.
> … <#141cce19f823cad9_>
> -- Jake Biesinger Graduate Student Xie Lab, UC Irvine
> On Fri, Oct 18, 2013 at 11:42 AM, anbangx notifications@github.com
> wrote: In
> genomix/genomix-data/src/main/java/edu/uci/ics/genomix/config/GenomixJobConf.java:
> 
> > @@ -384,16 +384,16 @@ private void fillMissingDefaults() { > > //
> > scaffolding > if (getInt(SCAFFOLDING_MIN_TRAVERSAL_LENGTH, -1) == -1) > -
> > setInt(SCAFFOLDING_MIN_TRAVERSAL_LENGTH, 20); Test comments in github —
> > Reply to this email directly or view it on GitHub<
> > https://github.com/uci-cbcl/genomix/pull/2/files#r7070360> .
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/uci-cbcl/genomix/pull/2/files#r7070393
> .

## 

Best Regards,

Anbang Xu

**Comment by jakebiesinger** [comment]
this is another major comment with task list.

We mustn't forget to
- [ ] go to the mall and buy Christmas presents.

Notice that if you click the above item, the markdown changes from [ ] to [x].  Nifty!

**Comment by jakebiesinger** [comment]
Good work, everybody. I will buy you Christmas presents :gift:

File: genomix/genomix-data/src/main/java/edu/uci/ics/genomix/type/DIR.java (hunk 2 above)
Comment (anbangx) [inline genomix/genomix-data/src/main/java/edu/uci/ics/genomix/type/DIR.java:15]: Great, it works. :smile: 

Try TODO List:
- [] @JavierJia finish this TODO list
- [] @jakebiesinger clean up code
- [] @Nan-Zhang please fix this issue #4 

``` ruby
require 'redcarpet'
markdown = Redcarpet.new("Hello World!")
puts markdown.to_html
```

Here's an example:

```
function test() {
  console.log("notice the blank line before this function?");
}
```

~~commend out~~

Comment (anbangx) [inline genomix/genomix-data/src/main/java/edu/uci/ics/genomix/type/DIR.java:15]: TODO List:
- [ ] @JavierJia finish this TODO list
- [ ] @jakebiesinger clean up code
- [ ] @Nan-Zhang please fix this issue #4

**Pull request merged**